internal/cmd/controller/reconciler/bundledeployment_controller.go

```diff
@@ -68,7 +69,13 @@ func (r *BundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			return err
 		}
 		t.Status = bd.Status
-		return r.Status().Update(ctx, t)
+		err = r.Status().Update(ctx, t)
+		if err != nil {
+			return err
+		}
+
+		metrics.CollectBundleDeploymentMetrics(t)
```

Comment:
Wouldn't it be simpler to leave metrics collection out of the retry loop, and to run it right before the final `return` (~L83), which would have the same effect?